test/IECore/CancellerTest.py

```diff
@@ -123,5 +124,19 @@ def raiser( key ) :
 
 			c.get( "c++" )
 
+	def testElapsedTime( self ) :
+
+		c = IECore.Canceller()
+		self.assertEqual( c.elapsedTime(), 0.0 )
```

Comment:
What precision does elapsedTime have?  The intent here is to show that the timer won't start until cancelled is called, but in order for the test to show that, we must assume that the timer precision is smaller than the time to execute a line of Python?  Probably true, but not obvious.

Actually, maybe it would be good to actually test that the units are seconds, by testing that the elapsedTime is approximately equal to 0.25, not just >=0.25, since that would pass for any small units?  Maybe this is so obvious it doesn't need testing, but I find the API to std::chrono pretty awful for this sort of thing.

Reply:
In libstdc++, `steady_clock::duration` is specified in nanoseconds, but I don't think the underlying measurements actually have nanosecond resolution. Nevertheless, calling `elapsedTime()` immediately in `cancel()` on the C++ side never gives me zero, so I think there's more than enough effective precision for this test where we also have python bindings in the way. I've added in a minimal sleep as an added precaution.

I had wanted to test approximate equality rather than just `>=`, but that usually ends up in adding some huge epsilon because the CI machines are under so much load at times. So I settled for checking it manually on my machine. But I've added a very crude ballpark check to the tests for a little additional protection.